src/Web/VKHS/Login.hs

```diff
@@ -117,7 +121,12 @@ initialAction = do
             , ("display", "wap")
             , ("response_type", "token")
             ])
-  return (DoGET u (cookiesCreate ()))
+  cookies <- liftIO $ do
+    c <- doesFileExist l_cookies_file
```

Comment:
That is not obvious, but l_access_token_file is de-facto a Maybe type: it will not attempt FS read/write if passed an empty string. Should be a FIXME actually. Could you please do nothing if `l_cookies_file` is empty?